lib/elbas/aws/autoscale_group.rb

```diff
@@ -17,12 +17,13 @@ def instances
       end
 
       def launch_template
-        raise Elbas::Errors::NoLaunchTemplate unless aws_counterpart.launch_template
+        lts = launch_template_specification
```

Comment:
Regarding my comment below, maybe:

```
lts = aws_launch_template || aws_launch_template_specification
raise Elbas::Errors::NoLaunchTemplate unless lts
```

Then split the method on L21 into those two parts, each one returning `nil` if not found? That way it is clear here what we are doing, getting one or the other, both of them following the same pattern (returning `nil` when nothing exists), and then this method raises in th at case since it cannot build the `LaunchTemplate`.

Reply:
Yes, I've made the changes following the pattern. The implementation(&.) requires ruby 2.3+ to work.